src/EditorFeatures/CSharp/GoToDefinition/CSharpGoToDefinitionSymbolService.cs

```diff
@@ -16,5 +16,38 @@ protected override ISymbol FindRelatedExplicitlyDeclaredSymbol(ISymbol symbol, C
         {
             return symbol;
         }
+
+        protected override bool TokenIsPartOfDeclaringSyntax(SyntaxToken token)
```

Comment:
❓ How do you feel about simplifying this down to just `ISyntaxFactsService.IsDeclaration` for now? While it doesn't cover parameters, local variables, or local functions, it provides a notable advantage: it's kept up-to-date with respect to the language and certainly keeps things here lightweight and simple.